RecoPixelVertexing/PixelTriplets/src/CACut.h

```diff
@@ -0,0 +1,175 @@
+// -*- C++ -*-
+// //
+// // Package:    RecoPixelVertexing/PixelTriplets
+// // Class:      CACut
+// //
+// // Original Author:  Karla Josefina Pena Rodriguez
+// //         Created:  Wed, 14 Feb 2019 10:30:00 GMT
+// //
+
+#ifndef RecoPixelVertexing_PixelTriplets_src_CACut_h
+#define RecoPixelVertexing_PixelTriplets_src_CACut_h
+
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "CAGraph.h"
+
+class CACut {
+public:
+  explicit CACut(const double defaultCut, const std::vector<edm::ParameterSet> &tripletCuts)
+      : useCACuts(true), foundAllLayerIds(false), defaultCut_(defaultCut) {
+    if (tripletCuts.size() == 1 && tripletCuts[0].getParameter<double>("cut") == -1.) {
+      useCACuts = false;
+      //TODO Uncomment the Following warning after tuning
+      //edm::LogWarning("Configuration") << "No CACut VPSet. Using default cut value of " << defaultCut << " for all layer triplets";
+      return;
+    }
+
+    valuesByTripletNames_.reserve(tripletCuts.size());
+    valuesByLayerIds_.reserve(tripletCuts.size());
+
+    setCutValuesByTripletNames(tripletCuts);
+  }
+
+  void setCutValuesByTripletNames(const std::vector<edm::ParameterSet> &tripletCuts) {
+    for (const auto &thisTriplet : tripletCuts) {
+      valuesByTripletNames_.emplace_back();
+      auto &thisCACut = valuesByTripletNames_.back();
+
+      thisCACut.tripletName = thisTriplet.getParameter<std::string>("seedingLayers");
+      thisCACut.cutValue = thisTriplet.getParameter<double>("cut");
+    }
+  }
+
+  void setCutValuesByLayerIds(CAGraph &caLayers) {
+    if (!useCACuts || foundAllLayerIds)
```

Comment:
This check shouldnt be needed:
- ```useCACuts``` is prevented being "false" from line 24
- ```foundAllLayerIds``` is initialized as "false" in the constructor

Reply:
Do we agree that this check should stay, to avoid the layer name -> layer id mapping when it's not needed?

Reply:
This comment was wrong, indeed: not to be taken into account